### PR TITLE
Added Jacoco support and tests.

### PR DESCRIPTION
**Pull request by nbantar**

### CODE REVIEW

File: src/main/java/docbuddy/users/persistence/DatastoreManager.java [hunk 1]
```diff
@@ -3,12 +3,20 @@
...
+@Setter
+@Repository
 public class DatastoreManager {
     private Datastore datastoreClient;
 
     public DatastoreManager() {
         this.datastoreClient = DatastoreOptions.getDefaultInstance().getService();
     }
+
+    public DatastoreManager(Datastore client) {
```
Comment (nestorreloaded) [inline src/main/java/docbuddy/users/persistence/DatastoreManager.java]: Shoul't we use Lombok @AllArgsConstructor?

File: src/main/java/docbuddy/users/persistence/dao/UserDaoImpl.java [hunk 2]
```diff
@@ -134,4 +154,14 @@ public User login(User user) {
         }
         return resultUsers;
     }
+
+    private void validateUser(User user) {
+        if (user.getUserName() == null || user.getUserName().isEmpty()) {
```
Comment (nestorreloaded) [inline src/main/java/docbuddy/users/persistence/dao/UserDaoImpl.java]: Maybe we can use Google Guava or Apache Commons String validation, they are already null zafe:

https://stackoverflow.com/questions/14721397/checking-if-a-string-is-empty-or-null-in-java

File: src/main/java/docbuddy/users/persistence/DatastoreManager.java (hunk 1 above)
Comment (nbantar) [reply src/main/java/docbuddy/users/persistence/DatastoreManager.java]: Good catch. Done.

File: src/main/java/docbuddy/users/persistence/dao/UserDaoImpl.java (hunk 2 above)
Comment (nbantar) [reply src/main/java/docbuddy/users/persistence/dao/UserDaoImpl.java]: Absolutely right. Done.

**Pull request merged**